File: user_interface/lib/presentation/pages/profile/widgets/plan_theme.dart
import 'package:flutter/material.dart';
import 'package:lefture/domain/plan_features.dart' as plan_features;
import 'package:lefture/presentation/themes/app_colors.dart';

/// プランごとのテーマカラー。tier_level(0=Free, 1=Lite, 2=Core, 3=Max)だけで
/// 判定する — 以前はplanName/storeProductIdの文字列に'premium'/'max'等が
/// 含まれるかで判定していたが、プラン名は今後マーケティング都合で変わり得る一方
/// tier_levelはDB上の安定した整数なので、判定はすべてこちらに統一した。
Color planThemeColor(int tierLevel) {
  switch (tierLevel) {
    case plan_features.tierMax:
      return const Color(0xFF7C4DFF);
    case plan_features.tierCore:
      return const Color(0xFFE11D48);
    case plan_features.tierLite:
      return AppColors.cosmicBlue;
    default:
      return AppColors.starGold;
  }
}

/// plan_icons/ 配下のアセットファイル名(拡張子込み)。判定方針はplanThemeColorと同じ
/// (tier_levelのみ)。
String planIconAsset(int tierLevel) {
  switch (tierLevel) {
    case plan_features.tierMax:
      return 'assets/images/plan_icons/galaxy.png';
    case plan_features.tierCore:
      return 'assets/images/plan_icons/solarsystem.png';
    case plan_features.tierLite:
      return 'assets/images/plan_icons/planet.png';
    default:
      return 'assets/images/plan_icons/stardust.png';
  }
}

File: webapp/src/lib/planTheme.ts
/**
 * プランのtier_level(0=Free, 1=Lite, 2=Core, 3=Max)からテーマカラーを決める。
 * plan_theme.dart の planThemeColor と、plan_card.dart の isPremium/isStandard
 * 出し分け(Max=紫のネオン、Core=紅のネオン)を1つにまとめたもの。
 * CreditsPage・PlansPageの両方で同じ色を使うための共有先。
 */
export interface TierTheme {
  accent: string;
  isPremium: boolean;
  isStandard: boolean;
}

export function tierAccent(tierLevel: number): TierTheme {
  if (tierLevel >= 3) return { accent: '#C084FC', isPremium: true, isStandard: false };
  if (tierLevel === 2) return { accent: '#FB7185', isPremium: false, isStandard: true };
  if (tierLevel === 1) return { accent: '#42A5F5', isPremium: false, isStandard: false };
  return { accent: '#FFB300', isPremium: false, isStandard: false };
}

/** planIconAsset(plan_theme.dart) 準拠。public/img/plan_icons/ に同名png済み。 */
export function planIconAsset(tierLevel: number): string {
  if (tierLevel >= 3) return '/img/plan_icons/galaxy.png';
  if (tierLevel === 2) return '/img/plan_icons/solarsystem.png';
  if (tierLevel === 1) return '/img/plan_icons/planet.png';
  return '/img/plan_icons/stardust.png';
}
